compiler/z/codegen/SystemLinkage.cpp

```diff
@@ -931,10 +932,21 @@ TR::S390zLinuxSystemLinkage::S390zLinuxSystemLinkage(TR::CodeGenerator * codeGen
    setReturnAddressRegister (TR::RealRegister::GPR14 );
 
    setGOTPointerRegister   (TR::RealRegister::GPR12  );
+
+#if defined(OMRZTPF)
+/*
+ * z/TPF os is a 64 bit target
+ */
+   setOffsetToRegSaveArea     (16);    // z/TPF is a 64 bit target only
+   setGPRSaveAreaBeginOffset  (16);    //x'10'  see ICST_R2 in tpf/icstk.h
+   setGPRSaveAreaEndOffset    (160);   //x'a0'  see ICST_PAT in tpf/icstk.h
+   setOffsetToFirstParm       (448);   //x'1c0' see ICST_PAR in tpf/icstk.h
+#else
```

Comment:
Thanks @jjohnst-us . I will update the comment to note the difference in stack sizes between z/tpf and z/linux and remove the first line of code; z/tpf can and does use the 64bit setting so no need to have that. The other three lines use values that directly reference our stack and need to be there or z/tpf can't start a jvm properly.